### PR TITLE
Add RTL support

### LINKED ISSUE #670: Add RTL support
https://en.wikipedia.org/wiki/Bi-directional_text

Thanks.

### PR DESCRIPTION
**Pull request by ImangazalievM**
Resolves #670

Copied from https://github.com/codex-team/editor.js/pull/909

### CODE REVIEW

**Comment by talyguryn** [comment]
Link for testing: https://deploy-preview-1248--editor-js.netlify.app/example-dev-rtl.html

**Comment by ImangazalievM** [comment]
1.I fixed the positioning of icons in the toolbar.
2. Unfortunately, there is nothing I can do about text selection, this is the standard behavior when mixing RTL and LTR text

**Review by talyguryn — APPROVED** [review]

**Review by ranemihir — APPROVED** [review]

File: example/example-dev-rtl.html [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  var editor = new EditorJS({
+    /**
+     * Wrapper of Editor
+     */
+    holder: 'editorjs',
+
+    /**
+     * Text direction
+     */
+    direction: 'rtl',
```
Comment (neSpecc) [inline example/example-dev-rtl.html:92]: let's move this option to the `i18n` section, near `messages`

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: example/example-dev-rtl.html [hunk 2]
```diff
@@ -0,0 +1,233 @@
+<!--
```
Comment (neSpecc) [inline example/example-dev-rtl.html:1]: let's name this file `example-rtl.html`

**Review by neSpecc — CHANGES_REQUESTED** [review]

**Review by neSpecc — CHANGES_REQUESTED** [review]
There is an issue with arrow navigation between blocks. And there is no ability to move caret by arrows if it is at the start or at the end of the block.

For now, it checks the end of the block for `right` arrow and the start of the block for `left` arrow.

See blockEvents.ts `arrowRightAndDown` and `arrowLeftAndUp`

File: src/components/__module.ts [hunk 3]
```diff
@@ -46,4 +46,12 @@ export default class Module {
   public set state(Editor: EditorModules) {
     this.Editor = Editor;
   }
+
+  /**
+   * Returns true if current direction is RTL (Right-To-Left)
+   */
+  protected isRtl(): boolean {
```
Comment (neSpecc) [inline src/components/__module.ts:53]: maybe it would better to use getter instead of method?

File: src/styles/rtl.css [hunk 4]
```diff
@@ -0,0 +1,74 @@
+.codex-editor.rtl-fix {
```
Comment (gohabereg) [inline src/styles/rtl.css:1]: Why is it called `fix`? Maybe just add a modifier: `.codex-editor--rtl`

**Review by neSpecc — COMMENTED** [review]
Almost ready, but it still has some issues:

In the "narrow mode" (which enables when editor inserted in small containers) there are some issues. Try to remove underscore from the `ce-example__content--small` class at example-rtl.html.

The narrow mode is used in small containers, where is no place for the Plus button and for the Block Settings button. It have two features:
- the plus button is placed inside the editor column

image

- additional padding-right added for the block settings button

image

We need to adopt these moments for the RTL

**Review by neSpecc — APPROVED** [review]

**Review by gohabereg — APPROVED** [review]

**Review by talyguryn — APPROVED** [review]

**Comment by javadbat** [comment]
can we use it now?

**Pull request merged**